### ISSUE TITLE
Backspace removes one too many parenthesis

### ISSUE THREAD

**Issue by MrAndersen1** [microsoft/vscode#108794]
Issue Type: <b>Bug</b>

When you have an if-statement with a function call in it and press backspace to remove the functions parenthesis it removes the closing parenthesis of the if-statement too.
```ts
declare function someFunctionOrVariable(): boolean;
if (someFunctionOrVariable()) { // put the cursor between the two closing parenthesis and press backspace twice
}
```
[Playground Link](https://www.typescriptlang.org/play?ts=4.0.2#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwGccBbEAMTU11QHkYA1WLKAIwhAAoBKALnhZw52UVAG4AUFkTwORUhXTY8dRjGZtOXLvADe8APT74AB2QZ4GABYIwyGERj8QGAO4gQ+Kwlc54kHARYqADmJrAeXoEE8CLAJnAE0SxQYADWBMYp3i5YYCDiAL5AA)

Context: I have gotten into the habit of testing poorly named boolean variables/methods by adding a pair of parenthesis behind it to see if it is a function. When it isn't and I remove the parenthesis the if-statement breaks too.

Rant (no need to read): For me this would not be an issue if autocomplete inserted parenthesis too if it is a function. While this is not always wanted it probably is most of the time? Or if the popup was triggered after autocomplete so I can see what it is. Or I simply just learn to read the popup help before I press enter to autocomplete :) . Or learn / rebind that cmd + k + i shortcut.
Still feels like it shouldn't remove the final parenthesis though :)

VS Code version: Code 1.50.1 (d2e414d9e4239a252d1ab117bd7067f125afd80a, 2020-10-13T14:53:05.704Z)
OS version: Darwin x64 19.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-1030NG7 CPU @ 1.10GHz (8 x 1100)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (1.87GB free)|
|Process Argv|-psn_0_323663 --crash-reporter-id 44640369-2a58-4ef9-8279-cc7b9068d6d4|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-pdf|yza|1.4.4

(2 theme extensions excluded)

</details>

**Comment by hediet** [comment]
This seems to be fixed now, I cannot reproduce it in VS Code 1.55. Please reopen if it is not fixed!

**Comment by MrAndersen1** [comment]
Great, it seems to be fixed for me too.
